File: backend 7.0/Models/Investment.cs
public class Investment
{
    public string Id { get; set; }
    public int UserId { get; set; }
    public string Type { get; set; }
    public string TransactionType { get; set; }

    // Stock fields
    public string StockName { get; set; }
    public string DematAccount { get; set; }
    public DateTime? PurchaseDate { get; set; }
    public int? NumberOfShares { get; set; }
    public double? Brokerage { get; set; }
    public string BrokerageType { get; set; }
    public double? PurchasePrice { get; set; }

    // Bond fields
    public string FixedIncomeName { get; set; }
    public DateTime? InvestmentDate { get; set; }
    public double? InvestmentAmount { get; set; }
    public double? CouponRate { get; set; }
    public string CompoundingFrequency { get; set; }
    public string InterestType { get; set; }
    public DateTime? MaturityDate { get; set; }

    // Mutual fund fields
    public string SchemeName { get; set; }
    public string FolioNumber { get; set; }
    public string AmountType { get; set; }
    public double? Amount { get; set; }
    public double? Price { get; set; }

    // Redemption/sell info
    public DateTime? RedemptionDate { get; set; }
    public string SellAt { get; set; }
    public double? SellPrice { get; set; }
}

File: src/app/core/models/investment.ts
export interface Investment {
  id: number;
  userId: number;
  type: 'stock' | 'bond' | 'mutualFund' | 'goldBond';
  transactionType: 'buy' | 'sell';

  // Common fields
  investmentDate?: string;
  purchasePrice?: number;
  quantity?: number;
  currentPrice?: number;
  currentValue?: number;

  // Stock-specific fields
  stockName?: string;
  dematAccount?: string;
  numberOfShares?: number;
  brokerage?: number;
  brokerageType?: '%' | 'rs';

  // Bond-specific fields
  fixedIncomeName?: string;
  investmentAmount?: number;
  rateOfInterest?: number; // Coupon Rate
  compoundingFrequency?:
    | 'monthly'
    | 'quarterly'
    | 'half-yearly'
    | 'yearly'
    | 'on maturity';
  interestType?: 'accrued' | 'interest payout';
  maturityDate?: string;

  // Mutual Fund-specific fields
  schemeName?: string;
  folioNumber?: string;
  amount?: number;
  units?: number;
  price?: number;

  // Gold Bond-specific fields
  securityName?: string;
  purchaseDate?: string;
  couponRate?: number;

  sellPrice?: number;
  redemptionDate?: string;
  sellAt?: string;
}
